src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1504,13 +1509,25 @@ public function getContent(bool $asResource = false)
         return $this->content;
     }
 
+    /**
+     * Gets the decoded form or json request body.
+     */
+    public function getPayload(): InputBag
+    {
+        return $this->request->count() ? clone $this->request : new InputBag($this->toArray());
+    }
+
     /**
      * Gets the request body decoded as array, typically from a JSON payload.
      *
      * @throws JsonException When the body cannot be decoded to an array
      */
     public function toArray(): array
     {
+        if (null !== $this->arrayContent) {
```

Comment:
Agree this method is a bit superfluous now. I guess if the request has a json body and `$_POST` parameters it could still be useful to get the json body? I don't know if that's a realistic scenario.

Reply:
Deprecations come at a cost, there is no need to remove this method IMHO.

Reply:
Think $_POST and JSON body combined is not possible via a real HTTP call? As $_POST is using the body to provide the data and so the body can not be used for json?

Reply:
> Think $_POST and JSON body combined is not possible via a real HTTP call? As $_POST is using the body to provide the data and so the body can not be used for json?

If that's true than the method is indeed superfluous. But, I agree with Nicolas, let's leave that to another PR.